src/main/java/christmas/util/InputValidator.java

```diff
@@ -0,0 +1,103 @@
+package christmas.util;
+
+import christmas.consts.Menu;
+import christmas.consts.SimpleConsts;
+import christmas.consts.Splitter;
+import christmas.domain.MenuOrderAssembler;
+import christmas.exception.DayExceptionCaller;
+import christmas.exception.OrderExceptionCaller;
+
+import java.util.EnumMap;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class InputValidator {
+    private final MenuOrderAssembler menuOrderAssembler;
+
+    public InputValidator(MenuOrderAssembler menuOrderAssembler) {
+        this.menuOrderAssembler = menuOrderAssembler;
+    }
+
+    public int parseToValidatedDate(String inputtedNumber) {
+        int validatedNumber = 0;
+        try {
+            validateEmpty(inputtedNumber);
+            validatedNumber = validateNumber(inputtedNumber);
+        } catch (IllegalArgumentException e) {
+            DayExceptionCaller.throwDayException(e);
+        }
+        validateMemorySafe(validatedNumber);
+        return validatedNumber;
+    }
+
+    private void validateEmpty(String inputtedString) {
+        if (inputtedString == null || inputtedString.isEmpty()) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    private int validateNumber(String inputtedNumber) {
+        int validatedNumber = 0;
+        try {
+            validatedNumber = Integer.parseInt(inputtedNumber);
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException(e);
+        }
+        return validatedNumber;
+    }
```

Comment:
저는 변수의 수는 무조건 적을 수록 좋다고 생각하는 편입니다.
어짜피 별다른 처리 없이 바로 리턴될 값을 지역변수로 추출해 둔다면, 이후 리팩토링을 하는 등의 상황에서 값을 변경할 수 있는 가능성이 열린다고 생각해요.

해당 메서드의 `validateNumber`라는 지역변수 역시 곧바로 리턴될 값이기에 지역변수를 만들지 않을 수 있다고 생각합니다.
```suggestion
    private int validateNumber(String inputtedNumber) {
        try {
            return Integer.parseInt(inputtedNumber);
        } catch (NumberFormatException e) {
            throw new IllegalArgumentException(e);
        }
    }
```

Reply:
try문 안에서 return을 하는 구조를 활용하니 불필요한 지역변수를 제거할 수 있었군요!
저도 불필요한 지역변수를 어쩔 수 없이 선언하고 있었는데, 알려주셔서 감사드립니다 :)